### DISCUSSION TITLE
outHumAbs in der Belchertown skin analog https://www.woellsdorf-wetter.de

### DISCUSSION THREAD

**Discussion by hoetzgit** [roe-dl/weewx-GTS#14] [Q&A] [answered]
Hallo Johanna,

ich würde gerne die Anzeige der relativen / absoluten Luftfeuchte im Block "Station observations" (wie auf Deiner Seite) auch in meiner Seite an dieser Stelle einbinden. Ich habe in `belchertown.py` die Anzeige-Logik "`rainWithRainRate`" für "`outHumidity`" angepasst und benutzt, aber mir wird dann natürlich die absolute Feuchte in der "`microgram_per_meter_cubed`" Einheit angezeigt. 

Hast Du einen Tipp für mich, wie ich den Wert an der Stelle in "`gram_per_meter_cubed`" ausgeben kann? Ich bekomme das selbst nicht hin.

Vielen Dank
Henry

**Comment by roe-dl** [comment]
`belchertown.py` habe ich wie folgt erweitert
```
            if obs=='outHumidity':
                humabs_output = getattr(current,'outHumAbs',None)
                if humabs_output is not None:
                    humabs_output = humabs_output.gram_per_meter_cubed
                    station_obs_html += "&nbsp;<span class='border-left'>&nbsp;</span>"
                    station_obs_html += '<span class="outHumAbs">%s</span><!-- AJAX -->' % humabs_output
```
Das steht am Schluß vor den abschließenden `</td>` `</tr>`

Die Variable `station_observations` enthält nur `outHumidity`. `outHumAbs` wird dort gar nicht erwähnt.

In `weewx.conf` muß ergänzt werden:
```
    [[MQTT]]
        ...
        [[[inputs]]]
            ...
            [[[[outHumAbs]]]]
                units = gram_per_meter_cubed
```

**Comment by hoetzgit** [comment]
Vielen Dank für Deine Unterstützung, die Einheit im `[[MQTT]]` Teil anzupassen hat sehr weitergeholfen. Der Report wird jetzt korrekt erstellt.

Hast Du noch irgendwo etwas anpassen müssen? Wenn eine MQTT Aktualisierung kommt, habe ich noch folgendes Verhalten: Der Wert wird nicht korrekt gerundet und es wird noch immer die Einheit '`microgram_per_meter_cubed`' angezeigt.

Wenn ich in die `weewx_data.json` sehe, dann ist `unit_rounding.outHumAbs` noch auf `0` und `unit_label.outHumAbs` noch auf "`µg/m³`"
Bei Deiner `weewx_data.json`  ist es korrekt.

In der `extension.py` habe ich folgendes eingetragen, obwohl es in Deiner Extension ja bereits definiert ist:
`weewx.units.default_unit_format_dict['gram_per_meter_cubed'] = '%.1f'`

Ohne Erfolg. Ich stehe auf dem Schlauch, hast Du noch den letzten Tipp für mich?

Vielen herzlichen Dank
Henry

**Comment by roe-dl** [comment, marked as answer]
In der Tat, da gab es noch eine Änderung in `belchertown.py`. Die hatte ich schon wieder vergessen.

Im Abschnitt "Get all observations and their rounding values" ganz am Schluß vor Beginn des nächsten Abschnitts und außerhalb der `for`-Schleife:
```
        all_obs_rounding_json["outHumAbs"] = 1
        all_obs_unit_labels_json["outHumAbs"] = " g/m³"
```

**Comment by hoetzgit** [comment]
Vielen Dank, so funktioniert es.

Viele Grüße
Henry
